### ISSUE TITLE
Posting 4MB images causes OutOfMemoryException

### ISSUE THREAD

**Issue by cpfeiffer** [mattermost/mattermost-mobile#1157]
#### Summary
Selecting a 4MB image from the camera or media store fails with an OutOfMemoryException.

#### Environment Information

- Device Name: Samsung S7, others
- OS Version: 5.0 - 7.1 (tested)
- Mattermost App Version: 1.4.0
- Mattermost Server Version: 4.3.1

#### Steps to reproduce
Use mattermost-mobile on Android and try to post a 4MB file. Take this one for example: http://www.letsgodigital.org/images/producten/1822/testrapport/panasonic-tz5-image.jpg

#### Expected behavior
Posting the image should work without a problem.

#### Observed behavior
You will get an OutOfMemoryException like this:
```
W/System.err( 5496): java.lang.OutOfMemoryError: Failed to allocate a 8386572 byte allocation with 8283224 free bytes and 7MB until OOM
W/System.err( 5496):    at java.io.ByteArrayOutputStream.expand(ByteArrayOutputStream.java:91)
W/System.err( 5496):    at java.io.ByteArrayOutputStream.write(ByteArrayOutputStream.java:201)
W/System.err( 5496):    at android.graphics.Bitmap.nativeCompress(Native Method)
W/System.err( 5496):    at android.graphics.Bitmap.compress(Bitmap.java:1011)
W/System.err( 5496):    at com.imagepicker.utils.MediaUtils.getResizedImage(MediaUtils.java:162)
W/System.err( 5496):    at com.imagepicker.ImagePickerModule.onActivityResult(ImagePickerModule.java:453)
W/System.err( 5496):    at com.facebook.react.bridge.ReactContext.onActivityResult(ReactContext.java:256)
W/System.err( 5496):    at com.facebook.react.ReactInstanceManager.onActivityResult(ReactInstanceManager.java:656)
W/System.err( 5496):    at com.reactnativenavigation.react.NavigationReactGateway.onActivityResult(NavigationReactGateway.java:94)
W/System.err( 5496):    at com.reactnativenavigation.controllers.NavigationActivity.onActivityResult(NavigationActivity.java:187)
```
or this:
```
I/art     ( 1993): Clamp target GC heap from 267MB to 256MB
I/art     ( 1993): Alloc concurrent mark sweep GC freed 3(96B) AllocSpace objects, 0(0B) LOS objects, 1% free, 251MB/256MB, paused 1.403ms total 46.081ms
I/art     ( 1993): Clamp target GC heap from 267MB to 256MB
E/art     ( 1993): Throwing OutOfMemoryError "Failed to allocate a 5861516 byte allocation with 4274626 free bytes and 4MB until OOM"
D/AndroidRuntime( 1993): Shutting down VM
D/ERROR   ( 1993): java.lang.OutOfMemoryError: Failed to allocate a 5861516 byte allocation with 4274626 free bytes and 4MB until OOM
D/ERROR   ( 1993):      at java.io.ByteArrayOutputStream.toByteArray(ByteArrayOutputStream.java:122)
D/ERROR   ( 1993):      at com.imagepicker.utils.MediaUtils.getResizedImage(MediaUtils.java:182)
D/ERROR   ( 1993):      at com.imagepicker.ImagePickerModule.onActivityResult(ImagePickerModule.java:453)
D/ERROR   ( 1993):      at com.facebook.react.bridge.ReactContext.onActivityResult(ReactContext.java:256)
D/ERROR   ( 1993):      at com.facebook.react.ReactInstanceManager.onActivityResult(ReactInstanceManager.java:656)
```

#### Possible fixes
One fix is this:
https://github.com/react-community/react-native-image-picker/pull/727
but it's not enough.

react-native-image-picker always recodes the picked image because you pass `0.7` as requested quality:
- https://github.com/mattermost/mattermost-mobile/blob/master/app/components/post_textbox/components/attachment_button.js#L26
- https://github.com/mattermost/mattermost-mobile/blob/master/app/components/post_textbox/components/attachment_button.js#L61
- https://github.com/mattermost/mattermost-mobile/blob/master/app/components/post_textbox/components/attachment_button.js#L96

The fix is to not recode the image by passing 1.0 as quality.

**Comment by enahum** [comment]
Hi @cpfeiffer this is great, do you think you can make a PR for this?

**Comment by enahum** [comment]
Closed via #1169
